### PR TITLE
glue together firm module and binary generation CLI

### PR DESCRIPTION
**Pull request by problame**

### CODE REVIEW

**Comment by problame** [comment]
bors try

**Comment by reiner-dolp** [comment]
the only thing missing from #134 is [commit 959ffdbfa](https://github.com/comprakt/comprakt/pull/134/commits/959ffdbfae9f76aa2424bbdaae41dd2149857f62). Will add this in a later commit since it conflicts heavly with what we have implemented in this PR.

**Comment by problame** [comment]
I 🍒 ed that one!

**Comment by reiner-dolp** [comment]
oh, thanks!

**Comment by reiner-dolp** [comment]
ah, but it didn't work. https://github.com/comprakt/comprakt/pull/137/files#diff-3fccd914d4013b283c3a1df6d630e173R64 should be something collision free ('.main')

**Comment by reiner-dolp** [comment]
@problame mjrt-rs fails to compile in release mode:
```
thread 'cli_ast_0_empty_class_mj' panicked at 'called `Result::unwrap()` on an `Err` value: CargoError { cause: Some(CargoError { kind: CommandFailed, context: Some("   Compiling mjrt-src v0.1.0 (/home/travis/build/comprakt/comprakt/mjrt-src)\n   Compiling mjrt-bin v0.1.0 (/home/travis/build/comprakt/comprakt/mjrt-bin)\nerror: failed to run custom build command for `mjrt-bin v0.1.0 (/home/travis/build/comprakt/comprakt/mjrt-bin)`\nprocess didn\'t exit successfully: `/home/travis/build/comprakt/comprakt/target/release/build/mjrt-bin-25aded626b1728f1/build-script-build` (exit code: 101)\n--- stderr\nthread \'main\' panicked at \'libmjrt_src-*.a not exactly one match: []\', mjrt-bin/build.rs:32:9\nnote: Run with `RUST_BACKTRACE=1` for a backtrace.\n\nwarning: build failed, waiting for other jobs to finish...\nerror: build failed\n"), cause: None }) }', libcore/result.rs:1009:5
```

**Comment by problame** [comment]
so what broke the tests, exactly?

**Comment by problame** [comment]
bors try

File: compiler-cli/src/main.rs [hunk 1]
```diff
@@ -284,6 +235,124 @@ macro_rules! setup_io {
...
+    // contains a file stem.
+    let out_name = input.file_stem().unwrap();
+
+    let out_path = if let Some(out_dir) = output {
+        out_dir.join(out_name)
+    } else {
+        PathBuf::new().join(out_name)
+    };
+
+    // compile runtime and link it with user's code
```
Comment (problame) [inline compiler-cli/src/main.rs]: inaccurate: we don't compile the runtime anymore

**Comment by reiner-dolp** [comment]
retrying with warm cache:

bors try

**Comment by problame** [comment]
Looks like the hack described in https://github.com/comprakt/comprakt/pull/137/files#diff-58ceb2fdbcf4280270d9729caaef9a85R4 doesn't work after all. I'll fix that.

**Comment by problame** [comment]
I verified that https://github.com/comprakt/comprakt/pull/137/commits/14f5b2082fccb2aae3c4295782f497e327028ab0 does not depend on what we have in the build cache by building it in a clean /tmp checkout.

@reiner-dolp  feel free to merge if the tests pass!

**Comment by problame** [comment]
bors try

File: mjrt/Cargo.toml [hunk 2]
```diff
@@ -1,14 +1,13 @@
 [package]
-name = "mjrt-bin"
+name = "mjrt"
 version = "0.1.0"
 authors = ["Christian Schwarz <me@cschwarz.com>"]
 edition = "2018"
-links = "mjrt_src"
+links = "mjrt"
```
Comment (flip1995) [inline mjrt/Cargo.toml]: shouldn't this be `mjrt_impl`?

Comment (problame) [reply mjrt/Cargo.toml]: I think we could just remove this line, we don't rely on the `DEP_MJRT_.*` environment variable anymore.

**Comment by reiner-dolp** [comment]
attempting merge with cold/emptied cache.

bors r+

**Pull request merged**